### PR TITLE
CDAP-7085 Add the User Id when making stream write requests in DefaultStreamWriter

### PR DESCRIPTION
**Pull request by gokulavasan**
JIRA : https://issues.cask.co/browse/CDAP-7085

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/app/stream/DefaultStreamWriter.java [hunk 1]
```diff
@@ -136,6 +136,7 @@ private void write(String stream, ByteBuffer data, Map<String, String> headers)
     for (Map.Entry<String, String> header : headers.entrySet()) {
       requestBuilder.addHeader(stream + "." + header.getKey(), header.getValue());
     }
+    addUserIdHeader(requestBuilder);
```
Comment (sreevatsanraman) [inline cdap-app-fabric/src/main/java/co/cask/cdap/app/stream/DefaultStreamWriter.java]: This should be passed in only if authorization is enabled right?

Comment (gokulavasan) [inline cdap-app-fabric/src/main/java/co/cask/cdap/app/stream/DefaultStreamWriter.java]: That check was in the private method `addUserIdHeader` but I have now removed that method.

**Comment by gokulavasan** [comment]
@sreevatsanraman @chtyim Please review when you get a chance. Thanks!

**Comment by sreevatsanraman** [comment]
LGTM

**Pull request merged**